Jellyfin.Server.Implementations/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -11,17 +17,59 @@ namespace Jellyfin.Server.Implementations.Extensions;
 /// </summary>
 public static class ServiceCollectionExtensions
 {
+    private static IDictionary<string, JellyfinDbProviderFactory> GetSupportedDbProviders()
+    {
+        var items = new Dictionary<string, JellyfinDbProviderFactory>();
+        foreach (var providerType in AppDomain
+            .CurrentDomain
+            .GetAssemblies()
+            .SelectMany(f => f.GetTypes())
+            .Where(e => e.IsClass && typeof(IJellyfinDatabaseProvider).IsAssignableFrom(e)))
+        {
+            var keyAttribute = providerType.GetCustomAttribute<JellyfinDatabaseProviderKeyAttribute>();
+            if (keyAttribute is null || string.IsNullOrWhiteSpace(keyAttribute.DatabaseProviderKey))
+            {
+                continue;
+            }
+
+            var provider = providerType;
+            items[keyAttribute.DatabaseProviderKey] = (services) => (IJellyfinDatabaseProvider)ActivatorUtilities.CreateInstance(services, providerType);
+        }
+
+        return items;
+    }
+
     /// <summary>
     /// Adds the <see cref="IDbContextFactory{TContext}"/> interface to the service collection with second level caching enabled.
     /// </summary>
     /// <param name="serviceCollection">An instance of the <see cref="IServiceCollection"/> interface.</param>
+    /// <param name="configurationManager">The server configuration manager.</param>
     /// <returns>The updated service collection.</returns>
-    public static IServiceCollection AddJellyfinDbContext(this IServiceCollection serviceCollection)
+    public static IServiceCollection AddJellyfinDbContext(this IServiceCollection serviceCollection, IServerConfigurationManager configurationManager)
     {
+        var efCoreConfiguration = configurationManager.GetConfiguration<DatabaseConfigurationOptions>("database");
+        var providers = GetSupportedDbProviders();
+        JellyfinDbProviderFactory? providerFactory = null;
+
+        if (efCoreConfiguration is null)
+        {
+            // when nothing is setup via new Database configuration, fallback to SqLite with default settings.
+            efCoreConfiguration = new DatabaseConfigurationOptions()
+            {
+                DatabaseType = "SqLite",
```

Comment:
Small nit: SQLite is the right capitalization but we might want to make those strings all lowercase anyway.

Reply:
i opted to make the lookup case insensitive instead